### ISSUE TITLE
Difference in bins distribution between Matplotlib & Holoviews

### ISSUE THREAD

**Issue by mmolet** [holoviz/holoviews#5010]
#### ALL software version info
Python 3.7.4 
On iMac (21.5-inch, 2017)
Using IDLE

#### Description of expected behavior and the observed behavior

Problem is: Different bins distribution between Matplotlib & Holoviews is obtained.

#### Complete, minimal, self-contained example code that reproduces the issue

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from sklearn.datasets import load_wine

wine = load_wine()

print("Feature Names : ", wine.feature_names)
print("\nTarget Names : ", wine.target_names)

wine_df = pd.DataFrame(wine.data, columns = wine.feature_names)
wine_df["Target"] = wine.target
wine_df["Target"] = ["Class_1" if typ==0 else "Class_2" if typ==1 else "Class_3"  for typ in wine_df["Target"]]

print("\nDataset Size : ", wine_df.shape)

print(wine_df.head())

Target1=wine_df.query('Target == "Class_1"')
Target2=wine_df.query('Target == "Class_2"')
Target3=wine_df.query('Target == "Class_3"')

x = Target1['proline']
y = Target2['proline']
z = Target3['proline']

plt.hist(x, bins=20,histtype='bar',color='blue',alpha=0.7,label='Class_1')
plt.hist(y, bins=20,histtype='bar',color='red',alpha=0.7,label='Class_2')
plt.hist(z, bins=20,histtype='bar',color='orange',alpha=0.7,label='Class_3')
plt.xlabel('proline')
plt.ylabel('Frequency')
plt.title('Malic Acid Distribution')
plt.legend(frameon=False)
plt.tight_layout()
plt.savefig("Test", dpi=300)
plt.show()

import holoviews as hv
hv.extension('bokeh')
from bokeh.plotting import show
from holoviews import dim, opts
import hvplot.pandas
hist=wine_df.hvplot.hist(
                    y="proline",
                    by="Target",
                    width=600, height=400,
                    ylim=(0,16),
                    alpha=0.7,
                    bins=20,
                    ylabel="Frequency",
                    title="Malic Acid Distribution")
show(hv.render(hist))

#### Screenshots or screencasts of the bug in action
![Screen Shot 2021-07-10 at 7 16 14 AM](https://user-images.githubusercontent.com/47699196/125162663-d5775b80-e14e-11eb-9a62-018490c5a105.png)

**Comment by mmolet** [comment]
A possible solution is doing this: 

plt.hist([x, y, z], bins=20, histtype='step', stacked=False, fill=True,color=['blue', 'red', 'orange'],alpha=0.7,label=['Class_1', 'Class_2', 'Class_3'])

Someone pointed out that "Holoviews was doing all this in one call and hence the 20 bins were spread among the full data set. For matplotlib, I was calling hist 3 times so the bins were being set independently."
